test/parallel/test-runner-cli-concurrency.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const { deepStrictEqual, strictEqual } = require('node:assert');
+const { spawnSync } = require('node:child_process');
+const { readdirSync, writeFileSync } = require('node:fs');
+const { join } = require('node:path');
+const { beforeEach, test } = require('node:test');
+
+function createTestFile(name) {
+  writeFileSync(join(tmpdir.path, name), `
+    const fs = require('node:fs');
+
+    fs.unlinkSync(__filename);
+    setTimeout(() => {}, 1_000_000_000);
+  `);
+}
+
+beforeEach(() => {
+  tmpdir.refresh();
+  createTestFile('test-1.js');
+  createTestFile('test-2.js');
+});
+
+test('concurrency of one', () => {
+  const cp = spawnSync(process.execPath, ['--test', '--test-concurrency=1'], {
+    cwd: tmpdir.path,
+    timeout: common.platformTimeout(1000),
+  });
+
+  strictEqual(cp.stderr.toString(), '');
+  strictEqual(cp.error.code, 'ETIMEDOUT');
+  deepStrictEqual(readdirSync(tmpdir.path), ['test-2.js']);
+});
+
+test('concurrency of two', () => {
+  const cp = spawnSync(process.execPath, ['--test', '--test-concurrency=2'], {
```

Comment:
how does this test if the files really run concurrently? We have fixtures that create a mutex, can we use them? https://github.com/nodejs/node/blob/966e3d34936b28bfb31f00a6f862226baa3bf9d0/test/parallel/test-runner-concurrency.js#L89-L98

Reply:
also, better to test the `--test-concurrency` within https://github.com/nodejs/node/blob/966e3d34936b28bfb31f00a6f862226baa3bf9d0/test/parallel/test-runner-concurrency.js to avoid the two tests using this "mutex" in parallel (and it is related to this file)

Reply:
I now relaized this assersion. `deepStrictEqual(readdirSync(tmpdir.path), [])`
anyway we can still move the test into that file but not a blocker.
also, probably skip second test if ` os.availableParallelism() < 3`

Reply:
> how does this test if the files really run concurrently?

We know that they are not running sequentially because each file never finishes. Since we know that the tests aren't running sequentially, we just need to verify that the batch size is respected. Plus, this change is not introducing any new concurrency functionality, it's just assigning a different value using the existing functionality.